src/ipath/proto.rs

```diff
@@ -0,0 +1,31 @@
+use fileops::Fd;
+use ipath;
+use ipath::{user_info, base_info, control_data, cmd, cmd_data};
+use std::mem;
+use std::error::Error;
+
+// TODO: to borrow or to not to borrow.
+// For the sake of brevity, assume the latest driver/device is in use
+pub fn userint(fd: Fd, u_info: user_info, b_info: base_info) ->
+    (control_data, user_info, base_info) {
+  let mut b_info_clone = b_info.clone();
+  let cmd_union = cmd_data { user_info : u_info.clone() };
+
+  let mut driver_cmd = cmd { cmd_type : ipath::CMD_ASSIGN_CONTEXT,
```

Comment:
When I got down to this i had to do a double take because, with the "cmd" lowercase, i didn't recognize the syntax.